device/core/test/_device_client_test.js

```diff
@@ -0,0 +1,257 @@
+    it('calls the done callback with an Error object if the upload fails', function(done) {
+      var DummyBlobUploader = function () {
+        this.uploadToBlob = function(blobName, stream, streamLength, callback) {
+          callback(new Error('fake error'));
+        };
+      };
+
+      var client = new Client(new EventEmitter(), null, new DummyBlobUploader());
+      client.uploadToBlob('blobName', new stream.Readable(), 42, function(err) {
+        assert.instanceOf(err, Error);
+        done();
+      });
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_041: [The `uploadToBlob` method shall call the `done` callback no parameters if the upload succeeds.]*/
+    it('calls the done callback with no parameters if the upload succeeded', function (done) {
+      var DummyBlobUploader = function () {
+        this.uploadToBlob = function(blobName, stream, streamLength, callback) {
+          callback();
+        };
+      };
+
+      var client = new Client(new EventEmitter(), null, new DummyBlobUploader());
+      client.uploadToBlob('blobName', new stream.Readable(), 42, done);
+    });
+  });
+
+  describe('#on(\'message\')', function () {
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_002: [The ‘message’ event shall be emitted when a cloud-to-device message is received from the IoT Hub service.]*/
+    it('emits a message event when a message is received', function (done) {
+      var dummyTransport = new FakeTransport();
+      var client = new Client(dummyTransport);
+      client.on('message', function(msg) {
+        /*Tests_SRS_NODE_DEVICE_CLIENT_16_003: [The ‘message’ event parameter shall be a ‘Message’ object.]*/
+        assert.equal(msg.constructor.name, 'Message');
+        done();
+      });
+
+      dummyTransport.emit('message', new Message());
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_004: [The client shall start listening for messages from the service whenever there is a listener subscribed to the ‘message’ event.]*/
+    it('starts listening for messages when a listener subscribes to the message event', function () {
+      var dummyTransport = new FakeTransport();
+      sinon.spy(dummyTransport, 'enableC2D');
+      var client = new Client(dummyTransport);
+
+      // Calling 'on' twice to make sure it's called only once on the receiver.
+      // It works because the test will fail if the test callback is called multiple times, and it's called for every time the 'message' event is subscribed on the receiver.
+      client.on('message', function () { });
+      client.on('message', function () { });
+      assert.isTrue(dummyTransport.enableC2D.calledOnce);
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_005: [The client shall stop listening for messages from the service whenever the last listener unsubscribes from the ‘message’ event.]*/
+    it('stops listening for messages when the last listener has unsubscribed', function (testCallback) {
+      var dummyTransport = new FakeTransport();
```

Comment:
>dummyTransport [](start = 10, length = 14)

ah, you wrote this test and I copied it.  You're the source of the non-PC variable names.  tsk. tsk. tsk.